src/Entity/Media.php

```diff
@@ -194,8 +194,7 @@ public function preSave(EntityStorageInterface $storage) {
       // @TODO We might modify that in the future but let's leave it like this
       // for now.
       if ($this->{$destination_field} && $this->{$destination_field}->isEmpty() && ($value = $bundle->getType()->getField($this, $source_field))) {
-        $property_name = $this->{$destination_field}->first()->mainPropertyName();
-        $this->{$destination_field}->first()->{$property_name} = $value;
+        $this->{$destination_field}->setValue([$value]);
```

Comment:
I don't really get what this is doing, one thing you should improve is the first condition do a $this->hasField($destination_field). You could also just do a $this->set($destination_field, $value), the API will figure out the rest. And it would actually allow that getField() method to return anything.. a single value for the main property, an array of multiple values for one item, or even multiple items.